extends Node2D

signal change_stage(next)
signal start_conversation(lines, item)

var controller: Node = null
@export
var yuhina: Node = null:
	set(value):
		$God.yuhina = value
var selecting: bool = false

const yuhina_start_at: Vector2 = Vector2(160, 180)

var yuhina_and_god = [
	"[b][color=blue]ç¥æ§˜[/color][/b] ã‚„ã‚ã‚ˆãæ¥ãŸãªã€yuhina!",
	"[b][color=red]yuhina[/color][/b] ã‚ï¼ãã®å£°ã¯3å¹´å‰ã«æ­»ã‚“ã ã¯ãšã®ç¥æ§˜ã§ã¯â€¼ï¸",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã»ã»ã»ã»ã€é©šã„ãŸã‹ğŸ‘»",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã¡ãªã¿ã«ã“ã‚“ãªé•·ã„ã‚»ãƒªãƒ•ã ã£ã¦é•·ã•ã‚’æ°—ã«ã›ãšä¸€æ°—ã«å–‹ã‚Œã‚‹ã®ã¯Scratchã«ã¯ç„¡ç†ãªèŠ¸å½“ã ã‚ã†ã€‚ã©ã†ã ï¼",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã€ã€ã€ã¾ã‚ã€é™ç•Œã¯ã‚ã‚‹ãŒãªã€‚ãã‚Œã«ãƒœã‚¿ãƒ³ã‚’æŠ¼ã•ãªãã¦ã‚‚å‹æ‰‹ã«é€²ã‚€ãŒãªğŸ˜€",
	"[b][color=red]yuhina[/color][/b] ã§ã€ã“ã‚“ãªã¨ã“ã‚ã§ä½•ã‚’ã—ã¦ã„ã‚‹ã®ã§ã™ã‹ï¼ï¼Ÿ",
	"[b][color=blue]ç¥æ§˜[/color][/b] ãŠå‰ã«ã‚„ã£ã¦ã‚‚ã‚‰ã„ãŸã„ã“ã¨ãŒã‚ã‚‹[color=red]ä»¥ä¸‹çœç•¥[/color]",
]

# 2å›ç›®ä»¥é™ã®ã‚»ãƒªãƒ•
var yuhina_and_god2 = [
	"[b][color=blue]ç¥æ§˜[/color][/b] ã‚‚ã†ãŠå‰ã«å–‹ã‚‹ã“ã¨ã¯ãªã„ãª",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã‚¤ãƒ™ãƒ³ãƒˆã‚’ã‚¯ãƒªã‚¢ã—ã¦ã‹ã‚‰ã¾ãŸæ¥ã‚‹ãŒã‚ˆã„ã€‚"
]

var warp = [
	"ğŸ± ãƒ¯ãƒ¼ãƒ—ãªã®ãƒ‹ãƒ£ãƒ¼ï¼ˆé‹è³ƒã¯è‡ªå‹•å¼•ãè½ã¨ã—ãªã®ã§æ®‹é«˜ã«æ°—ã‚’ã¤ã‘ã¦ãã ã•ã„ï¼‰ï¼",
	"[b][color=red]yuhina[/color][/b] ã­ã€ã­ã“ã®åˆ†éš›ã§ï¼"
]

func _on_god_yuhina_in(first):
	if first:
		start_conversation.emit(yuhina_and_god, false)
	else:
		start_conversation.emit(yuhina_and_god2, false)


func _on_kitty_kitty_meets_yuhina():
	start_conversation.emit(warp, false)
	change_stage.emit(2)

func _on_bed_coming_to_bed():
	if not selecting:
		selecting = true
		var x = await controller.select_from('å¯ã‚‹', 'å¾¹å¤œã™ã‚‹')
		selecting = false
